geps/gep-1958.md

```diff
@@ -0,0 +1,44 @@
+# GEP-1958: TLS ALPN Routing
+Issue: #1958
+Status: Provisional
+## TLDR
+Add support for routing traffic based on the application protocols present in the "client hello" step of a TLS handshake.
+
+## Goals
+
+- Provide a way to route a TLS connection to a specific backend based on application protocols.
+- Define expected behavior for routing based on application protocols with respect to ordering and ambiguous matches.
```

Comment:
This PR does not do this (or not completely, rather). Is it intended to achieve this goal in this PR or in the future?

Specifically, the handling of the fact a request has multiple ALPNs which are negotiated by the backends, and we may have listeners.

For instance, what is the behavior of:

Client ALPN: `[a, b, c]`
Listeners: `[no ALPN match, match ALPN=b, match ALPN=c]`
Server selects ALPN `c`

And many similar cases